resolve no-var violations in javascripts/main.js

We have some very old ESLint rules that I'd like to clean up, and our use of `var` is one of them. ES6 introduces `let` and `const` as a way to make clearer when a variable can be used, and I'd like to reduce our use of `var` where we don't need it:

If you know a bit about JS you can help out by following these steps:

 - create a new branch from `gh-pages`
 - in `javascripts/main.js` add this comment at the top of the file, before any source code:

```js
/* eslint no-var: [ "error" ] */
```

 - in a terminal, run `npm run lint`, which should emit a bunch of violations:

```
$ npm run lint

> up-for-grabs.net@1.0.0 lint /Users/shiftkey/src/up-for-grabs.net
> eslint javascripts/*.js tests/**/*.js


/Users/shiftkey/src/up-for-grabs.net/javascripts/main.js
   15:3   error  Unexpected var, use let or const instead  no-var
   18:3   error  Unexpected var, use let or const instead  no-var
   ...
```

 - run `npm run lint-fix` to fix whatever can be automatically tidied up
 - review and address the remaining violations manually
 - run `npm run prettier-fix` to ensure the formatting of the changes is fine
 - [test the site locally](https://github.com/up-for-grabs/up-for-grabs.net#testing-the-site-locally)
 - commit the changes and open a PR
 - test out the Netlify deploy site (click the `Details` link) and confirm the client-side functionality is still working

<img width="709" src="https://user-images.githubusercontent.com/359239/66651591-c70eaf80-ec09-11e9-85bc-a8c58d81e2be.png">

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -227,7 +227,7 @@ define([
     }
 
     if (!gh) {
-      countspantitle = $(
+      count = $(
         '<span class="count" title="Issue count is only available for projects on GitHub.">?</span>'
       ).appendTo(a);
       return;

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -219,7 +219,7 @@ define([
     const a = $(project).find('.label a');
       const gh = a
         .attr('href')
-        .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/),
+        .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/);
       counthref = a.find('.count');
 
     if (counthref.length) {

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -216,7 +216,7 @@ define([
   };
 
   const issueCount = function(project) {
-    const a = $(project).find('.label a'),
+    const a = $(project).find('.label a');
       gh = a
         .attr('href')
         .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/),

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -1,6 +1,7 @@
 // @ts-nocheck
 /* eslint prefer-arrow-callback: [ "error" ] */
 /* eslint arrow-parens: [ "error", "as-needed" ] */
+/* eslint no-var: [ "error" ] */
 
 define([
   'jquery',
@@ -13,10 +14,10 @@ define([
   // selector, and needs to be ready before this code runs
   'chosen',
 ], ($, loadProjects, ProjectsService, fetchIssueCount, _, sammy) => {
-  var compiledtemplateFn = null,
+  let compiledtemplateFn = null,
     projectsPanel = null;
 
-  var getFilterUrl = function() {
+  const getFilterUrl = function() {
     return location.href.indexOf('/#/filters') > -1
       ? location.href
       : `${location.href}filters`;
@@ -25,13 +26,13 @@ define([
   // inspired by https://stackoverflow.com/a/6109105/1363815 until I have a better
   // idea of what we want to do here
   function relativeTime(current, previous) {
-    var msPerMinute = 60 * 1000;
-    var msPerHour = msPerMinute * 60;
-    var msPerDay = msPerHour * 24;
-    var msPerMonth = msPerDay * 30;
-    var msPerYear = msPerDay * 365;
+    const msPerMinute = 60 * 1000;
+    const msPerHour = msPerMinute * 60;
+    const msPerDay = msPerHour * 24;
+    const msPerMonth = msPerDay * 30;
+    const msPerYear = msPerDay * 365;
 
-    var elapsed = current - previous;
+    const elapsed = current - previous;
 
     if (elapsed < msPerMinute) {
       return `${Math.round(elapsed / 1000)} seconds ago`;
@@ -52,7 +53,7 @@ define([
     return `about ${Math.round(elapsed / msPerYear)} years ago`;
   }
 
-  var renderProjects = function(projectService, tags, names, labels) {
+  const renderProjects = function(projectService, tags, names, labels) {
     const allTags = projectService.getTags();
 
     projectsPanel.html(
@@ -147,7 +148,7 @@ define([
     it fit URL specification
     @return string - The value of the Name
   */
-  var preparePopTagName = function(name) {
+  let preparePopTagName = function(name) {
     if (name === '') return '';
     return name.toLowerCase().split(' ')[0];
   };
@@ -156,9 +157,9 @@ define([
    * This is a utility method to help update URL Query Parameters
    * @return string - The value of the URL when adding/removing values to it.
    */
-  var updateQueryStringParameter = function(uri, key, value) {
-    var re = new RegExp('([?&])' + key + '=.*?(&|$)', 'i');
-    var separator = uri.indexOf('?') !== -1 ? '&' : '?';
+  let updateQueryStringParameter = function(uri, key, value) {
+    const re = new RegExp('([?&])' + key + '=.*?(&|$)', 'i');
+    const separator = uri.indexOf('?') !== -1 ? '&' : '?';
     if (uri.match(re)) {
       return uri.replace(re, '$1' + key + '=' + value + '$2');
     }
@@ -173,10 +174,10 @@ define([
    *
    * @return string - value of url params
    */
-  var getParameterByName = function(name, url) {
+  const getParameterByName = function(name, url) {
     if (!url) url = window.location.href;
     name = name.replace(/[\[\]]/g, '\\$&');
-    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
+    const regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
       results = regex.exec(url);
     if (!results) return null;
     if (!results[2]) return '';
@@ -189,7 +190,7 @@ define([
    * Also has corresponding fade-in and fade-out fetaure
    */
   $(window).scroll(() => {
-    var height = $(window).scrollTop();
+    const height = $(window).scrollTop();
     if (height > 100) {
       $('#back2Top').fadeIn();
     } else {
@@ -210,29 +211,29 @@ define([
    * @params String text - The text given, indices or names. As long as it is a string
    * @return Array - Returns an array of splitted values if given a text. Otherwise undefined
    */
-  var prepareForHTML = function(text) {
+  const prepareForHTML = function(text) {
     return text ? text.toLowerCase().split(',') : text;
   };
 
-  var issueCount = function(project) {
-    var a = $(project).find('.label a'),
+  const issueCount = function(project) {
+    const a = $(project).find('.label a'),
       gh = a
         .attr('href')
         .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/),
-      count = a.find('.count');
+      counthref = a.find('.count');
 
-    if (count.length) {
+    if (counthref.length) {
       return;
     }
 
     if (!gh) {
-      count = $(
+      countspantitle = $(
         '<span class="count" title="Issue count is only available for projects on GitHub.">?</span>'
       ).appendTo(a);
       return;
     }
 
-    count = $(
+    countspanimg = $(
       '<span class="count"><img src="images/octocat-spinner-32.gif" /></span>'
     ).appendTo(a);
 
@@ -241,20 +242,20 @@ define([
 
     fetchIssueCount(ownerAndName, labelEncoded).then(
       resultCount => {
-        count.html(resultCount);
+        countspanimg.html(resultCount);
       },
       error => {
         const message = error.message ? error.message : error;
-        count.html('?!');
-        count.attr('title', message);
+        countspanimg.html('?!');
+        countspanimg.attr('title', message);
       }
     );
   };
 
   $(() => {
-    var $window = $(window),
+    const $window = $(window),
       onScreen = function onScreen($elem) {
-        var docViewTop = $window.scrollTop(),
+        const docViewTop = $window.scrollTop(),
           docViewBottom = docViewTop + $window.height(),
           elemTop = $elem.offset().top,
           elemBottom = elemTop + $elem.height();
@@ -266,7 +267,7 @@ define([
 
     $window.on('scroll chosen:updated', () => {
       $('.projects tbody:not(.counted)').each(function() {
-        var project = $(this);
+        const project = $(this);
         if (onScreen(project)) {
           issueCount(project);
           project.addClass('counted');
@@ -279,30 +280,30 @@ define([
 
     projectsPanel.on('click', 'a.remove-tag', function(e) {
       e.preventDefault();
-      var tags = [];
+      const tags = [];
       projectsPanel
         .find('a.remove-tag')
         .not(this)
         .each(function() {
           tags.push($(this).data('tag'));
         });
-      var tagsString = tags.join(',');
+      const tagsString = tags.join(',');
       window.location.href = '#/tags/' + tagsString;
     });
 
     loadProjects().then(p => {
-      var projectsSvc = new ProjectsService(p);
+      const projectsSvc = new ProjectsService(p);
 
-      var app = sammy(function() {
+      const app = sammy(function() {
         /*
          * This is the route used to filter by tags/names/labels
          * It ensures to read values from the URI query param and perform actions
          * based on that. NOTE: It has major side effects on the browser.
          */
         this.get(/\#\/filters/, () => {
-          var labels = prepareForHTML(getParameterByName('labels'));
-          var names = prepareForHTML(getParameterByName('names'));
-          var tags = prepareForHTML(getParameterByName('tags'));
+          const labels = prepareForHTML(getParameterByName('labels'));
+          const names = prepareForHTML(getParameterByName('names'));
+          const tags = prepareForHTML(getParameterByName('tags'));
           renderProjects(projectsSvc, tags, names, labels);
         });
 

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -222,7 +222,7 @@ define([
         .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/);
       counthref = a.find('.count');
 
-    if (counthref.length) {
+    if (count.length) {
       return;
     }
 

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -217,7 +217,7 @@ define([
 
   const issueCount = function(project) {
     const a = $(project).find('.label a');
-      gh = a
+      const gh = a
         .attr('href')
         .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/),
       counthref = a.find('.count');

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -247,7 +247,7 @@ define([
       error => {
         const message = error.message ? error.message : error;
         count.html('?!');
-        countspanimg.attr('title', message);
+        count.attr('title', message);
       }
     );
   };

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -242,7 +242,7 @@ define([
 
     fetchIssueCount(ownerAndName, labelEncoded).then(
       resultCount => {
-        countspanimg.html(resultCount);
+        count.html(resultCount);
       },
       error => {
         const message = error.message ? error.message : error;

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -217,10 +217,10 @@ define([
 
   const issueCount = function(project) {
     const a = $(project).find('.label a');
-      const gh = a
-        .attr('href')
-        .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/);
-      let count = a.find('.count');
+    const gh = a
+      .attr('href')
+      .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/);
+    let count = a.find('.count');
 
     if (count.length) {
       return;

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -246,7 +246,7 @@ define([
       },
       error => {
         const message = error.message ? error.message : error;
-        countspanimg.html('?!');
+        count.html('?!');
         countspanimg.attr('title', message);
       }
     );

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -233,7 +233,7 @@ define([
       return;
     }
 
-    countspanimg = $(
+    count = $(
       '<span class="count"><img src="images/octocat-spinner-32.gif" /></span>'
     ).appendTo(a);
 

--- a/javascripts/main.js
+++ b/javascripts/main.js
@@ -220,7 +220,7 @@ define([
       const gh = a
         .attr('href')
         .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/);
-      counthref = a.find('.count');
+      let count = a.find('.count');
 
     if (count.length) {
       return;

